### DISCUSSION TITLE
Use PAPERLESS_CONSUMER_RECURSIVE & PAPERLESS_CONSUMER_SUBDIRS_AS_TAGS parameter as a recursive tagging system

### DISCUSSION THREAD

**Discussion by Cartache** [jonaswinkler/paperless-ng#1073] [Feature Requests]
Hi,

Thank you for this excellent product which is working perfectly and smoothly for me!
However, I am still hesitating to complete abandon my previous solution due to the problem explained just hereunder.

I am trying to transfer my > 5000 notes from Evernote to Paperless-NG.
I would like to keep the existing tagging per note in Evenote in Paperless-NG.
By looking at the different parameters and option currently available I thought that a solution to my problem could be to:
1. Use the functionality to prevent importing duplicate document
2. Combine point 1. with "subdirs tagging" as well as "recursive tagging" parameter 

to tag already existing documents in Paperless-NG by successive import of the same documents.

As an example:
1. Evernote: I export my Evernote notes with the "Tag1" tag
2. Paperless: I put my documents in my import Paperless dir in a subdir "Tag1": \\My_Import_Dir\Tag1
3. Paperless: Paperless imports the documents a first time and tags them with "Tag1"
4. Evernote: I export my Evernote notes with tag "Tag2" tag that could also have "Tag1"
5. Paperless: I put my documents in my import Paperless dir in a subdir "Tag2": \\My_Import_Dir\Tag2
6. Paperless: Paperless imports the documents a first time and tags them with "Tag1" 

This way some documents having both tags in Evernote will have the same tags in Paperless-NG. 
Currently, if I understand it well Paperless-NG discard duplicate documents without further treatment.

Thank you for taking this into consideration.

Sined

**Comment by jonaswinkler** [comment]
Not exactly sure what you're looking for after reading this twice.

If you put a document into the folder /my_import_dir/tag1/tag2/document.pdf, paperless will add both tag1 and tag2 to the document (at least it should, IIRC).

**Comment by Cartache** [comment]
Hi,

Thank you for this rapid reaction. 
My point is that, currently, if you import an already existing document, nothing happens.
An improvement could be to not import the same document but apply the new tag for the same document.
So if I wan to apply 5 different tags to the same document, I would "reimport" 5 x the same document but each time in a different folder.
There is no possibility in Evernote to export the tags. So I planned to export tag by tag the same documents.

Let's assume I have in total 30 documents with 5 different tags. 10 documents have 3 tags (tag1, tag2,tag3), 10 other documents have 3 tags (tag2, tag3 and tag4), and 10 other documents have 3 tags (tag3, tag4, tag5). The process to apply all the tags would be:
- Start exporting the 10 documents with tag1
- Import these documents in Paperless, if some of those documents are already in Paperless, the import process should simply apply the tag1 (/ImportFolder/tag1/docX). 
- Export the documents with the tag2
- Import these documents in Paperless, if some of those documents are already in Paperless, the import process should simply apply the tag2 (/ImportFolder/tag2/docX). 
- ... For all the tags

This way all the documents will be imported with all the tags applied in different batches.

I hope this is clearer.

Sined
